pkgs/build-support/fetchurl/boot.nix

```diff
@@ -4,12 +4,16 @@ let mirrors = import ./mirrors.nix; in
 
 { url ? builtins.head urls
 , urls ? []
-, sha256
+, sha256 ? ""
+, hash ? ""
 , name ? baseNameOf (toString url)
 }:
 
+# assert only one hash is set
+assert hash != "" -> sha256 == "";
```

Comment:
Is there a reason for allowing `fetchurl` without any hashes? I figure that to preserve the old behaviour, the code should also assert that at least one hash is set (and use `null` as default value, to distinguish from an intentionally empty value).

Reply:
Last time I tried ofborg failed because of this evaluation, although I'm not able to reproduce this locally.

Reply:
Looks like I was too tired yesterday and forgot how boolean logic works :/ Pushed the change

Reply:
I see that the other fetchurl has `hash ? ""`, so I should drop my comment about `null`! All good for me.

Reply:
now it's not possible to set the hash to "" and TOFU

It works in fetchurl/default https://github.com/NixOS/nixpkgs/blob/e04579e7cdce875453574a46123b73dfe6db046f/pkgs/build-support/fetchurl/default.nix#L123-L134